src/Snap/Internal/Http/Server.hs

```diff
@@ -155,11 +164,13 @@ httpServe :: Int                         -- ^ default timeout
           -> [ListenPort]                -- ^ ports to listen on
           -> ByteString                  -- ^ local hostname (server name)
           -> Maybe (ByteString -> IO ()) -- ^ access log action
+          -> Maybe AccessLogHandler
           -> Maybe (ByteString -> IO ()) -- ^ error log action
+          -> Maybe ErrorLogHandler
           -> ([Socket] -> IO ())         -- ^ initialisation
           -> ServerHandler               -- ^ handler procedure
           -> IO ()
-httpServe defaultTimeout ports localHostname alog' elog' initial handler =
+httpServe defaultTimeout ports localHostname alog' alh elog' elh initial handler =
```

Comment:
Unfortunately since we're changing this signature on an exported module (even if it does live in .Internal), this will force a version bump to 1.0.